Vision renewed (#2300)
deepchecks/deepchecks: docs/source/user-guide/vision/supported_tasks_and_formats.rst

.. _supported_tasks_and_formats:

===========================
Supported Tasks and Formats
===========================

In order to analyze the images, labels and predictions provided in the
:doc:`VisionData object </user-guide/vision/vision_data_class>`,

> they must be provided the accepted deepchecks format. In this section we will describe the supported formats

Reviewer: ```suggestion
they must be provided in the accepted deepchecks format. In this section we will describe the supported formats
```

for each supported task type.

The VisionData object automatically validates your data formats when initialized, using the first batch of data. However,
in addition to the automatic validations, it is also important to make sure visually that the data was provided in the

> correct format via the :func:`head <deepchecks.vision.vision_data.VisionData.head>` functionality.

Reviewer: also this link is broken...
I think after API ref fix we have to manually check these as we don't receive warnings :-/

> correct format via the :func:`head <deepchecks.vision.vision_data.VisionData.head>` functionality.

Reviewer: ```suggestion
correct format via the :func:`VisionData.head() <deepchecks.vision.vision_data.VisionData.head>` functionality.
```

> .. note::

Reviewer: consider changing note to a different admonition (e.g. attention or something)

Also - maybe worth noting here about setting the reshuffle parameter or some link for more info (cause of the current status with the warning)

    In order to properly function, in addition for the data being in the correct format,
    the data must also be loaded in a **shuffled** manner.

We will first describe the formats which are common for all tasks, and then describe the label and
prediction formats which vary between tasks.

Common Formats
==============

> Batch General Format

Reviewer: Can we for these three "common formats" can we also give "examples" of "how it may look", like for the task-specific (predictions, labels...)?
I think the visuals can really help understand the meaning while reading the words

Reviewer: It can also have an explanation above/after 
(e.g. "where 100 is the number of samples, x is ... and ...")

--------------------
Each batch, which is the output of the :doc:`batch loader </user-guide/vision/vision_data_class>`,

> is required to be a :class:`dictionary <deepchecks.vision.vision_data.BatchOutputFormat>`

Reviewer 2: @shir22 Couldn't make this link work

Reviewer 2: pending api ref fix

Image identifiers need to be provided as an iterable of strings.

.. _supported_tasks__classification:
Classification
==============
Label Format
------------
Classification label per sample is expected to be provided as either a string or an int
representing the class index. For example, labels for a batch containing 2 samples can be provided as:

.. code-block:: python

    [1, 3]

Prediction Format
-----------------
Classification prediction for each sample should be the **prediction probabilities** per class, meaning that it should be an
iterable of floats in the range [0, 1] with length equal to the number of classes.

For example, predictions of a batch containing 2 samples for a classification task with 3 possible
classes may look like this:

.. code-block:: python

    [
        [0.1, 0.2, 0.7],
        [0.3, 0.6, 0.1]
    ]

.. _supported_tasks__object_detection:
Object Detection
================
Label Format
------------
Object detection label per sample should be an iterable of bounding boxes. Each bounding box should be an iterable
of 5 elements in the following order: ``(class_id, x_min, y_min, w, h)``.

>     ``x_min`` and ``y_min`` are the coordinates (in pixels) of the **top left corner** of the bounding box, ``w``

Reviewer: are this (and the one below, in the prediction format) indented on purpose?
this is how it looks
![220125 at 12 58 49 screenshot Arc](https://user-images.githubusercontent.com/33841818/214546268-b7b4f231-a8eb-441a-9c0c-5a34b3495412.png)

Reviewer 2: It's the same in label format, and I think it's good

Reviewer: Yes mentioned it was the same on both (and put here the printscreen from the other one lol)
just wondered if it was meant to be a "note" or something like that.
I would consider making them bullets, may be clearer/more aesthetical
anyway not critical

Reviewer 2: I think that bullets would be an overkill here.

.. _supported_tasks_and_formats:

===========================
Supported Tasks and Formats
===========================

In order to analyze the images, labels and predictions provided in the
:doc:`VisionData object </user-guide/vision/VisionData>`,
they must be provided in the accepted deepchecks format. In this section we will describe the supported formats
for each supported task type.

The VisionData object automatically validates your data formats when initialized, using the first batch of data. However,
in addition to the automatic validations, it is also important to make sure visually that the data was provided in the
correct format via the :func:`VisionData.head() <deepchecks.vision.vision_data.VisionData.head>` functionality.

.. admonition:: Shuffling your Data
   :class: attention
   In order to properly function, in addition for the data being in the correct format,

>    the data must also be loaded in a **shuffled** manner. For When using python data loaders, deepchecks will by default

Reviewer: did you mean PyTorch DataLoaders (instead of python data loaders...?)

>    attempt re-shuffle the data. However, if you are using any other framework, you must make sure to shuffle the data

Reviewer: ```suggestion
   attempt to reshuffle the data. However, if you are using any other framework, you must make sure to shuffle the data
```

>    yourself before passing it to deepchecks. The behaviour of re-shuffling the data can be controlled via the

Reviewer: ```suggestion
   yourself before passing it to deepchecks. The behaviour of reshuffling the data can be controlled via the
```

   ``reshuffle_data`` argument of the :class:`VisionData <deepchecks.vision.vision_data.VisionData>` constructor,

>    and it should be manually set to `False` for non-pytorch data.

Reviewer: ```suggestion
   and it should be manually set to `False` for non-PyTorch data.
```
